### PR TITLE
Migrate clinseq converge dgidb

### PR DESCRIPTION
**Pull request by gatoravi**
Move the dgidb converge scripts from the original scripts folder into ClinSeq/Command/Converge/ folder with a little bit of refactoring to work under the GMS. These are older scripts from @malachig.

### CODE REVIEW

File: lib/perl/Genome/Model/ClinSeq/Command/Converge/Base.pm [hunk 1]
```diff
@@ -716,8 +716,247 @@ sub add_read_counts{
...
+  foreach my $b (@$builds){
+    my $build_directory = $b->data_directory;
+    my $subject_common_name = $b->subject->common_name;
+    my $subject_name = $self->get_somatic_subject_common_name($b);
+    $subject_name =~ s/[\s-]/_/g;
+    my $build_id = $b->id;
+
+    #If the subject name is not defined, die
+    unless ($subject_name){
+      die $self->error_message("\n\nCould not determine subject name for build: $build_id\n\n");
```
Comment (tmooney) [inline lib/perl/Genome/Model/ClinSeq/Command/Converge/Base.pm]: I recommend removing the extra `\n`s from these messages.  They separate the message from the `ERROR:` that precedes them, which makes the logs more confusing to follow.

File: lib/perl/Genome/Model/ClinSeq/Command/Converge/Base.pm [hunk 2]
```diff
@@ -716,8 +716,247 @@ sub add_read_counts{
...
+    }
+
+    my $final_name = "Unknown";
+    if ($subject_name){$final_name = $subject_name;}
+    if ($subject_common_name){
+      $final_name = $subject_common_name . "_" . $subject_name;
```
Comment (tmooney) [inline lib/perl/Genome/Model/ClinSeq/Command/Converge/Base.pm]: Is it possible for there to be a `$subject_common_name` but no `$subject_name`?

File: lib/perl/Genome/Model/ClinSeq/Command/Converge/DgidbCounts.pm [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        print GENES "$patient\t$data_type\t$gene\n";
+      }
+    }
+  }
+  close TOTALS;
+  close GENES;
+}
+
+#Print summary of druggable genes broken down by event and data type
+sub write_summary1 {
```
Comment (tmooney) [inline lib/perl/Genome/Model/ClinSeq/Command/Converge/DgidbCounts.pm]: Is the `1` in this name significant?

File: lib/perl/Genome/Model/ClinSeq/Command/Converge/DgidbCounts.pm [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  my $k_g = $k_result->{'genes'}; #Known druggable genes
+  my $k_i = $k_result->{'interactions'}; #Known druggable gene interactions
+  my $data_type_sum = $k_result->{'data_type_sum'};
+  my (%totals, %data_type_sum_collapsed);
+  $self->write_summary1(\%totals, $data_type_sum, \%data_type_sum_collapsed, $k_g);
+  $self->write_totals_genes(\%totals);
+  return 1;
+}
+
+sub write_totals_genes {
```
Comment (tmooney) [inline lib/perl/Genome/Model/ClinSeq/Command/Converge/DgidbCounts.pm]: These subroutines may benefit from using `Genome::Utility::IO::SeparatedValueWriter`.

File: lib/perl/Genome/Model/ClinSeq/Command/Converge/DgidbCounts.pm (hunk 3 above)
Comment (gatoravi) [inline lib/perl/Genome/Model/ClinSeq/Command/Converge/DgidbCounts.pm]: depends on the null :P 
good catch, removing the 1.

File: lib/perl/Genome/Model/ClinSeq/Command/Converge/Base.pm [hunk 5]
```diff
@@ -483,7 +483,7 @@ sub getModelsBuilds{
     my @models = $mg->models;
     $target_count = scalar(@models);
   }
-  if ($verbose){Genome::Model->debug_message("\nSearching for $target_count models/builds");}
+  if ($verbose){$self->debug_message("\nSearching for $target_count models/builds");}
```
Comment (nnutter) [inline lib/perl/Genome/Model/ClinSeq/Command/Converge/Base.pm]: The newlines that @tmooney pointed out occur in many places not just the specific one he mentioned.

Comment (gatoravi) [inline lib/perl/Genome/Model/ClinSeq/Command/Converge/Base.pm]: ok, i removed a bunch here https://github.com/gatoravi/genome/commit/e6e508d6af775554f956d499642e6cc2d5f5f744, I'll work on removing the rest here. There might be a lot of files in ClinSeq with this issue, that might have to be a separate PR I think.

Comment (nnutter) [inline lib/perl/Genome/Model/ClinSeq/Command/Converge/Base.pm]: It can definitely be deferred to another PR.  I was only referring to lines you changed not all ClinSeq.  I realize you didn't add the newlines.

File: lib/perl/Genome/Model/ClinSeq/Command/Converge/DgidbCounts.pm (hunk 4 above)
Comment (gatoravi) [inline lib/perl/Genome/Model/ClinSeq/Command/Converge/DgidbCounts.pm]: i've updated these.

**Comment by nnutter** [comment]
:+1:

**Comment by tmooney** [comment]
+1, I'm glad we're getting those `.pl`s replaced! :sunny:
